File: server/src/main/java/com/example/hyundaiboot/dto/PushHistoryDto.java
package com.example.hyundaiboot.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.*;

import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@ToString
@Getter
@Setter
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class PushHistoryDto {
	String userId;
	String userName;
	String deviceId;
	String deviceDescription;
	String pushTitle;
	String pushContent;
	LocalDateTime pushDateTime;
	String sentState;
	LocalDateTime sentDateTime;
}

File: app/lib/dto/push_history_dto.dart
class PushHistoryDto {
  final String userId;
  final String userName;
  final String deviceId;
  final String deviceDescription;
  final String pushTitle;
  final String pushContent;
  final DateTime pushDateTime;
  final String sentState;
  final DateTime sentDateTime;

  const PushHistoryDto({
    required this.userId,
    required this.userName,
    required this.deviceId,
    required this.deviceDescription,
    required this.pushTitle,
    required this.pushContent,
    required this.pushDateTime,
    required this.sentState,
    required this.sentDateTime,
  });
}
